impeller/geometry/path.cc

```diff
@@ -61,6 +61,34 @@ void Path::SetConvexity(Convexity value) {
   convexity_ = value;
 }
 
+void Path::Shift(Point shift) {
+  size_t currentIndex = 0;
+  for (const auto& component : components_) {
+    switch (component.type) {
+      case ComponentType::kLinear:
+        linears_[component.index].p1 = linears_[component.index].p1 + shift;
```

Comment:
I don't know C++.

That being said, can you write:

```cc
liners[component.index].p1 += shift
```

...? or does that not work with vectors/arrays?

If not, at least pull `index` into `auto const index = component.index` :)

Reply:
Done

Reply:
Fun fact, += is a different operator than + (see point.h)